camlibs/sierra/library.c

```diff
@@ -460,7 +461,11 @@ sierra_write_packet (Camera *camera, char *packet, GPContext *context)
 			(camera->pl->flags & SIERRA_WRAP_USB_MASK),
 			packet, length));
 	} else {
-		CHECK (gp_port_write (camera->port, packet, length));
+		if (camera->pl->flags & SIERRA_NO_BLOCK_WRITE) {
+			for (x = 0; x < length; x++)
+				CHECK (gp_port_write(camera->port, &packet[x], 1));
+		} else
+			CHECK (gp_port_write (camera->port, packet, length));
```

Comment:
ultimately it's Marcus call but I find the 'optional' curly braces confusing. At least the `else` block should be in curly braces too.